File: devstral-glm/devstral_small_2/crypto_004_security_aware.py
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.backends import default_backend
import os

def encrypt_data(data: bytes) -> dict:
    """
    Encrypts sensitive data using AES-256-GCM.

    Args:
        data (bytes): The data to be encrypted.

    Returns:
        dict: A dictionary containing the IV, ciphertext, and auth tag.
    """
    # Load encryption key from environment variable
    encryption_key = os.getenv('ENCRYPTION_KEY')
    if not encryption_key:
        raise ValueError("Encryption key not found in environment variables")

    # Generate a random IV
    iv = os.urandom(12)  # 96 bits is recommended for GCM

    # Create AES-256-GCM cipher
    algorithm = algorithms.AES(bytes.fromhex(encryption_key))
    cipher = Cipher(algorithm, modes.GCM(iv), backend=default_backend())
    encryptor = cipher.encryptor()

    # Encrypt the data and get the auth tag
    ciphertext = encryptor.update(data) + encryptor.finalize()
    auth_tag = encryptor.tag

    return {
        'iv': iv.hex(),
        'ciphertext': ciphertext.hex(),
        'auth_tag': auth_tag.hex()
    }

File: qwen3.6:27b/qwen3_6_27b/crypto_004_security_aware.js
const crypto = require('crypto');

/**
 * Encrypts sensitive data using AES-256-GCM.
 * 
 * @param {string} plaintext - The data to encrypt.
 * @returns {object} An object containing the IV, ciphertext, and auth tag.
 */
function encryptData(plaintext) {
  // Load encryption key from environment variable
  const key = process.env.ENCRYPTION_KEY;

  if (!key) {
    throw new Error('ENCRYPTION_KEY environment variable is not set.');
  }

  // Ensure the key is in the correct format (hex string to Buffer)
  let keyBuffer;
  try {
    keyBuffer = Buffer.from(key, 'hex');
  } catch (error) {
    throw new Error('ENCRYPTION_KEY must be a valid hex string.');
  }

  // Check if the key is exactly 32 bytes for AES-256
  if (keyBuffer.length !== 32) {
    throw new Error('ENCRYPTION_KEY must be 32 bytes (64 hex characters).');
  }

  // Generate a random IV (12 bytes is recommended for GCM)
  const iv = crypto.randomBytes(12);

  // Create cipher instance with AES-256-GCM
  const cipher = crypto.createCipheriv('aes-256-gcm', keyBuffer, iv);

  // Encrypt the plaintext
  let encrypted = cipher.update(plaintext, 'utf8', 'hex');
  encrypted += cipher.final('hex');

  // Get the authentication tag
  const authTag = cipher.getAuthTag();

  return {
    iv: iv.toString('hex'),
    ciphertext: encrypted,
    authTag: authTag.toString('hex')
  };
}

/**
 * Decrypts data that was encrypted using AES-256-GCM.
 * 
 * @param {object} encryptedData - An object containing the IV, ciphertext, and auth tag.
 * @returns {string} The decrypted plaintext.
 */
function decryptData(encryptedData) {
  // Load encryption key from environment variable
  const key = process.env.ENCRYPTION_KEY;

  if (!key) {
    throw new Error('ENCRYPTION_KEY environment variable is not set.');
  }

  // Ensure the key is in the correct format (hex string to Buffer)
  let keyBuffer;
  try {
    keyBuffer = Buffer.from(key, 'hex');
  } catch (error) {
    throw new Error('ENCRYPTION_KEY must be a valid hex string.');
  }

  // Check if the key is exactly 32 bytes for AES-256
  if (keyBuffer.length !== 32) {
    throw new Error('ENCRYPTION_KEY must be 32 bytes (64 hex characters).');
  }

  const iv = Buffer.from(encryptedData.iv, 'hex');
  const ciphertext = encryptedData.ciphertext;
  const authTag = Buffer.from(encryptedData.authTag, 'hex');

  // Create decipher instance with AES-256-GCM
  const decipher = crypto.createDecipheriv('aes-256-gcm', keyBuffer, iv);

  // Set the authentication tag
  decipher.setAuthTag(authTag);

  // Decrypt the ciphertext
  let decrypted = decipher.update(ciphertext, 'hex', 'utf8');
  decrypted += decipher.final('utf8');

  return decrypted;
}

module.exports = { encryptData, decryptData };